cpp/include/cudf/detail/segmented_reduction_functions.hpp

```diff
@@ -0,0 +1,320 @@
+  device_span<size_type const> offsets,
+  data_type const output_dtype,
+  null_policy null_handling,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Computes sum of squares of elements of segments in the input column
+ *
+ * If input segment is empty, the segment result is null.
+ *
+ * If `null_handling==null_policy::INCLUDE`, all elements in a segment must be valid
+ * for the reduced value to be valid.
+ * If `null_handling==null_policy::EXCLUDE`, the reduced value is valid if any element
+ * in the segment is valid.
+ *
+ * @throw cudf::logic_error if input column type is not arithmetic type
+ * @throw cudf::logic_error if `output_dtype` is not floating point type
+ *
+ * @param col Input column data
+ * @param offsets Indices to identify segment boundaries within input `col`
+ * @param output_dtype Data type of the output column
+ * @param null_handling Specifies how null elements are processed for each segment
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return Column of `output_dtype` for the reduction results of the segments
+ */
+std::unique_ptr<column> segmented_sum_of_squares(
+  column_view const& col,
+  device_span<size_type const> offsets,
+  data_type const output_dtype,
+  null_policy null_handling,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Computes the standard deviation of elements of segments in the input column
+ *
+ * If input segment is empty, the segment result is null.
+ *
+ * If `null_handling==null_policy::INCLUDE`, all elements in a segment must be valid
+ * for the reduced value to be valid.
+ * If `null_handling==null_policy::EXCLUDE`, the reduced value is valid if any element
+ * in the segment is valid.
+ *
+ * @throw cudf::logic_error if input column type is not arithmetic type
+ * @throw cudf::logic_error if `output_dtype` is not floating point type
+ *
+ * @param col Input column data
+ * @param offsets Indices to identify segment boundaries within input `col`
+ * @param output_dtype Data type of the output column
+ * @param null_handling Specifies how null elements are processed for each segment
+ * @param ddof Delta degrees of freedom.
+ *             The divisor used is N - ddof, where N the number of elements in each segment
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return Column of `output_dtype` for the reduction results of the segments
+ */
+std::unique_ptr<column> segmented_standard_deviation(
+  column_view const& col,
+  device_span<size_type const> offsets,
+  data_type const output_dtype,
+  null_policy null_handling,
+  cudf::size_type ddof,
```

Comment:
Should we use the namespace? (Inconsistent with other parameters.)
```suggestion
  size_type ddof,
```